### ISSUE TITLE
ie11: layout-fill pushes elements outside of parent

### ISSUE THREAD

**Issue by epelc** [angular/material#1766]
`layout-fill` seems to be broken in some cases on ie11. It's use of `min-height: 100%;` seems to be causing elements to push other elements outside of their containers. However it works as expected if you switch `min-height: 100%;` to `height: 100%;`. Perhaps something like the `@-moz-document url-prefix()` but for ie would work as switching to `height: 100%;` in other browsers doesn't work.(edit: It appears to work in the latest chrome but I haven't tested thoroughly)

![image](https://cloud.githubusercontent.com/assets/5204642/6466684/4202e33e-c17e-11e4-8316-dd19b837ae70.png)
The `Welcome` header and signup/forgot password links are both within `md-card-content`

The markup is similiar to this
```
<div layout="row" layout-align="center center" flex class="login">
    <md-card layout="column" flex="25" flex-lg="40" flex-md="60" flex-sm="90">
        <md-card-content layout="column" layout-align="center center">
            <h1>Welcome</h1>
            <div layout-fill>
               <!-- Login form/google & facebook signin buttons -->
            </div>
            <small layout-margin layout="column" layout-align="center center">
                <!-- Signup/recovery links -->
            </small>
        </md-card-content>
    </md-card>
</div>
```

I think this should be confirmed on other versions of windows and along with older versions of ie but here is a nice stackoverflow answer with how to target specific versions of ie with css hacks similar to the firefox one. http://stackoverflow.com/a/27315792/2252814

Angular-material version: `0.8.2`
OS: `Windows 10 64bit Preview Build 9926`

**Comment by ThomasBurleson** [comment]
Please provide a CodePen or Plunkr that demonstrates this issue.

**Comment by epelc** [comment]
@ThomasBurleson I don't have access to a windows machine right now so I didn't get to test this. But It's essentially the same markup as what I showed in the picture. I'll try to test it in ie11 tomorrow to confirm it though.

http://codepen.io/anon/pen/ZYqMBw

**Comment by epelc** [comment]
@ThomasBurleson I just confirmed my demo above shows the issue in ie11 on windows 10.


It should look something like this
![image](https://cloud.githubusercontent.com/assets/5204642/6638610/318d381e-c95e-11e4-88a8-3f28324c413b.png)

However if you switch `min-height: 100%;` to `height: 100%;` as I said above for `layout-fill` it works perfectly. But this needs to only be applied in ie since `height: 100%;` gives very different results in chrome.

![image](https://cloud.githubusercontent.com/assets/5204642/6638688/f1a9e34a-c95e-11e4-9761-3d1c64b04409.png)

**Comment by ngraef** [comment]
Seems to be the same underlying IE bug as #946.

**Comment by gkalpak** [comment]
@epelc, I am confused be your use of `layout-fill`. Isn't it supposed to fill the parent ? If you are using it inside a parent with other elements, isn't it expected to "push them off the edge" ?

**Comment by epelc** [comment]
@gkalpak I wanted the username/password/login btn `div` to be full width. But I just removed the `layout-fill` from the `form` and it works now.

IE still seems to have issues with `layout-fill` but at least my card works now.

**Comment by gkalpak** [comment]
I somehow fill you are using `[layout-fill]` in place of `[flex]`. `[layout-fill]` is supposed to fill the parent and if it filles the parent, then there's no room for other elements, right ?
If you want to make it grow and take up all _extra_ space, the `[flex]` is what you need.

Do I miss something ?

**Comment by epelc** [comment]
@gkalpak I thought I should use `flex` too but it doesn't work because the div holding the login buttons is in a `layout="column"` soit flexes vertically.

I made my own `[full-wdth]` property which sets `width: 100%;`. It works pretty well for this case.

**Comment by gkalpak** [comment]
A, right. How about `align-self: stretch` (or even `align-items: stretch` on the parent if appropriate) ?
This way you can have one attribute for both horizontal and vertical layouts :wink:

**Comment by epelc** [comment]
@gkalpak Thanks I just switched `[full-width]` to use that instead.

**Comment by aaroncalderon** [comment]
I experienced a similar issue, but on chrome. In my case the parent element was div, so the top portion of my content was off the  window view area with no way to scroll up to be able to see it.

This only happens when the window is re-sized and the content is actually taller than the parent element. What I did is override the [layout-fill] with `height: auto;` to fix the issue. This is inside a conditional `@media` rule when this issue happens in my particular case, see below.

```css
@media screen and (max-width: 599px) {
  [layout-fill] {
    height: auto;
  }
}
```
